Fix ocean

gabrielfedel opened the pull request:
Change QE6500Class to OceanClass (manufacturer name), and add some small fixes on OceanClass

gabrielfedel committed:
Rename QE65000 to OceanClass. Rename all internal names to Ocean. This class works for all Ocean spectrometers, not only QE65000

gabrielfedel committed:
Fix way to choose between Spectra and DarkCorrectedSpetra in OceanClass

gabrielfedel committed:
On OceanClass Verify what spectrum before saveSpectrum

hhslepicka commented:
Hi @gabrielfedel, Why change the name? OceanOptics makes more than just one equipment and the model name is QE65000. I'm not sold on this change. Would you mind to clarify?

gabrielfedel commented:
Hi @hhslepicka , we will use the same class to another ocean optics spectrometer (they work very similar), so I think rename it is a good idea. 
Do you think let QE65000 is better?

hhslepicka commented on py4syn/epics/OceanClass.py:
```diff
@@ -100,6 +96,14 @@ def setCountStop(self):
+
+        # the spectra come from different pv if use darkcorrection
+        if dark == 1:
+            self.pvSpectrum = PV(self.pv+":DarkCorrectedSpectra")
```
Please instead of creating the PV in here, do it at __init__ and do get on the proper object.
This will mean that every time that you try to save a Spectrum it will try to create a new connection and check PyEpics cache and etc.

Change to something like this:
```python
if self.pvDarkCorrection.get() == 1:
    self.spectrum = self.pvDarkCorrectedSpectra.get(count=self.numPoints, as_numpy=True)
else:
    self.spectrum = self.pvSpectra.get(count=self.numPoints, as_numpy=True)
```

gabrielfedel replied:
I has to choose PV in saveSpectrum... So maybe is better I create both on init, and choose what use on saveSpectrum

hhslepicka replied:
That is exactly what I said! 👍

gabrielfedel replied:
Oh! sorry, I read only the text and no the code..hehe

I will perform this change

Thank you!

hhslepicka replied:
Awesome! I will wait for it to merge! Meanwhile also think about the name changing.

gabrielfedel replied:
:+1:

hhslepicka commented:
@gabrielfedel, are they using the same IOC?
If they differ somehow please create a class with the similarities and leave the specifics to the appropriated classes which inherit from this one.

gabrielfedel commented:
@hhslepicka Yes, they use the same ioc. I'm not sure yet the differences. Maybe I let QE65000 and when I notice the difference I create a subclass

hhslepicka commented:
@gabrielfedel What model are we talking about in here? Which IOC?

gabrielfedel commented:
QE65000 and HR2000, the same IOC manage both equipments.

hhslepicka commented:
@gabrielfedel so in this case a name such as OceanOpticsSpectrometer would be better.
And if you decide to go this way, please add to the docstrings that it supports those two models.
Please let me know what you decide on that.

gabrielfedel committed:
Changes on Ocean.py

Change class name to OceanOpticsSpectrometer
Added models tested with class
Move to init Spectrum PVs create.

gabrielfedel commented:
Done! :+1:

hhslepicka commented:
Thank you @gabrielfedel for the changes!
